packages/core-plugin-api/src/analytics/Tracker.ts

```diff
@@ -43,13 +61,69 @@ export class Tracker implements AnalyticsTracker {
       attributes,
     }: { value?: number; attributes?: AnalyticsEventAttributes } = {},
   ) {
+    // Never pass internal "_element" context value.
+    const { _element, ...context } = this.context;
+
+    // Never fire the special "_routable-extension-rendered" internal event.
+    if (action === '_routable-extension-rendered') {
+      // Instead, push it onto the global store.
+      routableExtensionRenders.push({
+        context: {
+          ...context,
+          extension: 'App',
+        },
+      });
+      return;
+    }
+
+    // If we are about to fire a real event, and we have an un-fired gathered
+    // mountpoint navigation on the global store, we need to fire the navigate
+    // event first, so this real event happens accurately after the navigation.
+    if (gatheredNavigations.length) {
+      // Combine the most recent info from each.
+      const lastGatheredNavigation = gatheredNavigations.pop()!;
+      const lastRoutableRender = routableExtensionRenders.pop();
+
+      try {
+        this.analyticsApi.captureEvent({
+          ...lastGatheredNavigation,
+          ...lastRoutableRender,
+        });
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Error during analytics event capture. %o', e);
+      }
+
+      // Clear the global stores.
+      gatheredNavigations.length = 0;
+      routableExtensionRenders.length = 0;
+    }
+
+    // Never directly fire a navigation event on a gathered route with default
+    // contextual details.
+    if (
+      action === 'navigate' &&
+      _element === 'gathered' &&
+      context.pluginId === 'root'
+    ) {
+      // Instead, push it onto the global store.
```

Comment:
Is there a risk that we have hold extension renders laying around at this point that would get incorrectly associated with this gathered navigation? i.e. should we clear `routableExtensionRenders` here?

Reply:
Think swapping this from an array to a single optional value clarifies what's going on here a bit. Lemme know if not!